app/assets/javascripts/reducers/users.js

```diff
@@ -23,12 +23,16 @@ const SORT_DESCENDING = {
 
 export default function users(state = initialState, action) {
   switch (action.type) {
-    case RECEIVE_USERS: return {
-      ...state,
-      users: action.data.course.users,
-      isLoaded: true,
-      lastRequestTimestamp: Date.now()
-    };
+    case RECEIVE_USERS: {
+      const sortedUsers = sortByKey(action.data.course.users, state.sort.sortKey, 'username');
```

Comment:
Per my earlier comment, I think this is where we probably want to use the `refresh` parameter of `sortByKey`. Otherwise, if the list gets sorted by `username` by the user, then when new user data is received, it will resort it by `username` but it will take `username` as the previous sort key as well, so it will reverse the sorting. I guess that could still result in changing the order of the list if the user has already sorted by something is a non-default direction, though, because sortByKey is implicitly written to assume that it's just acting on the already-received set of models and re-sorting them. hmm... not entirely sure what the best way around that issue is. but I think i'm okay with ignoring that problem, and just going with the `refresh` route so that in most cases it will behave as expected.

Reply:
Hi @ragesoss ,I tried adding the refresh parameter here but didn't quite see much change. Maybe it will need a change in the utils in the sortByKey function but I think the current issue only affects username and probably ignoring it won't be a bad idea?
Kindly let me know.

Reply:
I think the refresh parameter should definitely be used here (and there is no need to provide the previous key parameter, which can just be null).

Reply:
Okay, lemme implement it.

Reply:
Updated it.